File: backend/src/main/java/com/demo/aiknowledge/entity/Product.java
package com.demo.aiknowledge.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String productCode;
    private Long categoryId;
    private String title;
    private String brand;
    private String subCategory;
    private BigDecimal basePrice;
    private String imageUrl;
    private String description;
    private String tags;
    private BigDecimal rating;
    private Integer reviewCount;
    private Integer salesCount;
    private Integer status;
    private Integer embeddingStatus;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: android/app/src/main/java/com/evanyao/shopagent/data/model/Product.kt
package com.evanyao.shopagent.data.model

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class Product(
    @SerializedName(value = "id", alternate = ["product_id"])
    val id: Long,
    val productCode: String? = null,
    val categoryId: Long? = null,
    val title: String,
    val brand: String? = null,
    @SerializedName(value = "subCategory", alternate = ["sub_category"])
    val subCategory: String? = null,
    @SerializedName(value = "basePrice", alternate = ["price", "base_price"])
    val basePrice: BigDecimal,
    @SerializedName(value = "imageUrl", alternate = ["image_url"])
    val imageUrl: String? = null,
    val description: String? = null,
    val tags: String? = null,
    val rating: BigDecimal? = null,
    @SerializedName(value = "reviewCount", alternate = ["review_count"])
    val reviewCount: Int = 0,
    @SerializedName(value = "salesCount", alternate = ["sales_count"])
    val salesCount: Int = 0,
    val quantity: Int = 1,  // 购物车中数量（仅购物车列表使用）
    val status: Int = 1,
    val createTime: String? = null,
    val updateTime: String? = null
)
